src/index.ts

```diff
@@ -80,13 +80,22 @@ export const traceProvider: TraceProvider = {
   },
 
   /**
-   * @description Used to propagate traces. Accepts a serialised trace
-   * metadata string and returns a traceContext instance that can be passed
-   * to the wrapHandler.
+   * @description Used to propagate traces. Accepts serialised trace
+   * metadata and returns a traceContext instance that can be passed
+   * to wrapHandler.
    */
   deserializeTraceMetadata: async (
-    traceMetadata: string
-  ): Promise<unknown> => ({
-    distributedTraceHeaders: JSON.parse(traceMetadata),
-  }),
+    traceMetadata: unknown
+  ): Promise<unknown> => {
```

Comment:
Instead of unknown should this be an object?

Reply:
@sugendran From an interface point of view, steveo just needs to handle the metadata as an opaque value and pass it on the the next hook. After thinking about this I think the most semantically correct way may be to have the return type as `Promise<TraceContext>` where `type TraceContext = unknown;`
Does this sound sane to you?

Reply:
Update: As discussed: `type TraceContext = Record<string, string>;`